client.go

```diff
@@ -1147,10 +1170,12 @@ Loop:
 func (l *link) muxFlow() error {
 	// copy because sent by pointer below; prevent race
 	var (
-		linkCredit    = l.receiver.maxCredit - uint32(len(l.messages))
+		linkCredit    = l.receiver.maxCredit - uint32(len(l.pendingMessages))
```

Comment:
the linkcredit value that we send to the peer is calculated based on pendingMessage, not based on messages chan.